electrum/plugins/ledger/ledger.py

```diff
@@ -444,14 +504,80 @@ def sign_transaction(self, tx, password):
                 while inputIndex < len(inputs):
                     singleInput = [ chipInputs[inputIndex] ]
                     self.get_client().startUntrustedTransaction(False, 0,
-                                                            singleInput, redeemScripts[inputIndex], version=tx.version)
+                                                                singleInput, redeemScripts[inputIndex], version=tx.version)
                     inputSignature = self.get_client().untrustedHashSign(inputsPaths[inputIndex], pin, lockTime=tx.locktime)
                     inputSignature[0] = 0x30 # force for 1.4.9+
                     my_pubkey = inputs[inputIndex][4]
                     tx.add_signature_to_txin(txin_idx=inputIndex,
                                              signing_pubkey=my_pubkey.hex(),
                                              sig=inputSignature.hex())
                     inputIndex = inputIndex + 1
+
+
+                #TODO: strategy design pattern?
```

Comment:
should be implemented if needed

Reply:
`add_3k_signatures_to_tx` method extracted